lib/fritzbox/smarthome/actor.rb

```diff
@@ -4,36 +4,64 @@ class Actor < Resource
       include ActiveModel::Model
 
       attr_accessor \
-        :id,
-        :type,
-        :ain,
-        :present,
-        :name,
-        :manufacturer,
-        :group_members
+      :id,
+      :type,
+      :ain,
+      :present,
+      :name,
+      :manufacturer,
+      :group_members
+
+      ResourceNotFound = Class.new(RuntimeError)
 
       class << self
         def all(types: ['group', 'device'])
-          response = get(command: 'getdevicelistinfos')
-          xml = nori.parse(response.body)
+          xml = parse(get(command: 'getdevicelistinfos'))
+
           Array.wrap(types.map { |type| xml.dig('devicelist', type) }.flatten).compact.map do |data|
             klass = Actor.descendants.find { |k| k.match?(data) } || Actor
             self.in?([klass, Actor]) ? klass.new_from_api(data) : nil
           end.compact
         end
 
+        def find_by!(id: nil, ain: nil)
+          raise ArgumentError, 'Find via :id is not implmentet yet' if id.present?
+          raise ArgumentError, 'Missing argument :ain' if ain.nil?
```

Comment:
We can use Ruby mandatory keywords for this:

```suggestion
        def find_by!(ain:)
```

Reply:
True. I was already thinking ahead of cases where someone would like to fetch with the `:id` instead of the `:ain` - but I guess it's better to keep it for the future.